examples/publisher_untyped.rs

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+use iceoryx_rs::PublisherBuilder;
+use iceoryx_rs::Runtime;
+
+use bytes::BufMut;
+
+use std::error::Error;
+use std::thread;
+use std::time::Duration;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    Runtime::init("publisher_untyped");
+
+    let publisher = PublisherBuilder::<[u8]>::new("Radar", "FrontLeft", "Counter").create()?;
+
+    let mut counter = 0u32;
+    loop {
+        let sample = if counter % 2 == 1 {
+            // with initialized slice
+            let mut sample = publisher.loan_slice_with_alignment(
+                std::mem::size_of::<u32>(),
+                std::mem::align_of::<u32>(),
+            )?;
+            sample.as_mut().put_u32_le(counter);
+            sample
+        } else {
+            // with uninitialized slice
+            let mut sample = publisher.loan_uninit_slice_with_alignment(
+                std::mem::size_of::<u32>(),
+                std::mem::align_of::<u32>(),
+            )?;
+            unsafe {
+                sample.slice_assume_init_mut().put_u32_le(counter);
+                sample.assume_init()
```

Comment:
loan_slice_with_alignment seems sensible. 

loan_uninit_slice_with_alignment doesn't seem to really provide a significant distinction here compared to loan_slice_with_alignment, since loan_slice_with_alignment returns an uninitialized slice without requiring unsafe handling.  

---

The way I did it was also a little loose about initialization safety since it's always up to the user to populate the buffer with something meaningful before publishing regardless of how it's abstracted.  

The publisher could loan a PubChunkHandle which owns the data and exposes different views for convenience (depending on how the user wants to write into the buffer).  Regardless of which method is used, the PubChunkHandle owns the data and can publish it at will.

```rust
impl Publisher {
    pub fn loan_chunk(&self, size: usize) -> Result<PubChunkHandle, AllocationError>
}

pub struct PubChunkHandle<'a> {
    data: *mut ffi::c_void,
    publisher: &'a Publisher,
    sent: bool,
    size: usize,
}

impl PubChunkHandle<'_> {
    pub fn publish(mut self) 
    pub fn as_uninit_slice(&mut self) -> &mut UninitSlice 
    pub fn as_slice(&mut self) -> &mut [u8] 
    pub unsafe fn as_uninitialized<T>(&mut self) -> &mut std::mem::MaybeUninit<T> 
}
```

The last one is probably a bit sketchy since the user data could probably overflow, and it's easy to accidentally use a type the contains pointers.

Reply:
@neilisaac I think you overlooked the `.fill_with(|| T::default());` in `loan_slice_with_alignment` which initializes the slice. `loan_uninit_slice_with_alignment` doesn't do this but is unsafe. I tried to match the proposed API for uninitialized constructors of Box, Rc, and Arc.

I like the idea to expose a different view and will add this for `[u8]`
```rust
unsafe fn try_as_uninit<T: ShmSend>(&mut self) -> Option<&mut MaybeUninit<T>>
```
This would return `None` if the size and alignment do not match the requirements for `T`.

Reply:
@neilisaac I added the convenience functions and if you do not have any concerns, I would merge the PR